AirCasting/Utils/DownloadableImage.swift

```diff
@@ -0,0 +1,70 @@
+// Created by Lunar on 22/06/2022.
+//
+
+import SwiftUI
+import Combine
+
+struct DownloadableImage: View {
+    @StateObject var imageLoader: ImageLoader
+    
+    init(url: URL) {
+        _imageLoader = .init(wrappedValue: ImageLoader(url: url))
+    }
+    
+    var body: some View {
+        switch imageLoader.image {
+        case .success(let image):
+            Image(uiImage: image)
+                .resizable()
+                .scaledToFit()
+        case .failure:
+            VStack(alignment: .center) {
+                Image(systemName: "exclamationmark.triangle")
+                    .frame(width: 80, height: 80, alignment: .center)
+                    .font(.system(size: 30))
+                    .foregroundColor(.aircastingGray)
+                Text("Failed to load the image")
+            }
+        case .none:
+            ProgressView()
+                .progressViewStyle(CircularProgressViewStyle())
+                .padding(.vertical)
+        }
+    }
+}
+
+
+class ImageLoader: ObservableObject {
+    @Published var image: Result<UIImage, Error>?
+    
+    struct NoImageError: Error { }
+
+    init(url: URL) {
```

Comment:
this is a huge init, doing two separate things, splitting this into methods would bring more clarity here